### PR TITLE
Migrate Tapir from Polygon Mumbai to Polygon Amoy

### PR DESCRIPTION
**Pull request by derekpierre**
**Type of PR:**
- [ ] Bugfix
- [ ] Feature
- [ ] Documentation
- [X] Other

**Required reviews:** 
- [ ] 1
- [ ] 2
- [X] 3

**What this does:**

- Updated Tapir scripts to use Polygon Amoy (80002) instead of Polygon Mumbai (80001) which will be deprecated
- Deployed contracts from https://github.com/nucypher/nucypher-contracts/pull/242 (https://github.com/nucypher/nucypher-contracts/commit/3b684e5f4dd99887db5abab8f2a5a133776a292e) to Polygon Amoy for Tapir
- Updated Tapir contract registry based on latest deployments

**Issues fixed/closed:**
> - Fixes #...

Related to https://github.com/nucypher/nucypher/pull/3468.

**Why it's needed:**
> Explain how this PR fits in the greater context of the NuCypher Network.
> E.g., if this PR address a `nucypher/productdev` issue, let reviewers know!

**Notes for reviewers:**
> What should reviewers focus on? 
> Is there a particular commit/function/section of your PR that requires more attention from reviewers?

### CODE REVIEW

**Review by piotr-roslaniec — APPROVED** [review]

**Review by piotr-roslaniec — APPROVED** [review]

File: scripts/tapir/configure_staking.py [hunk 1]
```diff
@@ -45,10 +46,10 @@ def configure_sepolia_root(transactor: Transactor) -> int:
     return min_stake_size
 
 
-def configure_mumbai_root(transactor: Transactor, stake_size: int):
-    """Configures MockTACoApplication on Mumbai."""
+def configure_amoy_root(transactor: Transactor, stake_size: int):
+    """Configures MockTACoApplication on Amoy."""
     # Set up Tapir stakes on Mumbai
```
Comment (manumonti) [inline scripts/tapir/configure_staking.py:51]:
```suggestion
    # Set up Tapir stakes on Amoy
```

**Review by manumonti — CHANGES_REQUESTED** [review]
LGTM, but I caught a little comment that needs to be fixed I think

File: scripts/tapir/configure_staking.py (hunk 1 above)
Comment (derekpierre) [reply scripts/tapir/configure_staking.py]: Nice catch! Thanks!

**Review by manumonti — APPROVED** [review]
LGTM 🙌

**Pull request merged**